src/main/java/org/junitpioneer/jupiter/ExpectedToFailExtension.java

```diff
@@ -42,7 +43,12 @@ private static void invokeAndInvertResult(Invocation<Void> invocation, Extension
 				throw t;
 			}
 
-			String message = getExpectedToFailAnnotation(extensionContext).value();
+			ExpectedToFail expectedToFail = getExpectedToFailAnnotation(extensionContext);
+			if (Stream.of(expectedToFail.onExceptions()).noneMatch(c -> c.isInstance(t))) {
```

Comment:
Personally, I am not a fan of single letter lambda parameters but that is just personal preference. 🐸

Reply:
For me it is either way. Paramount is of course the overall code consistency.